drivers/timer/nrf_rtc_timer.c

```diff
@@ -117,19 +120,39 @@ void z_clock_set_timeout(s32_t ticks, bool idle)
 	ticks = MAX(MIN(ticks - 1, (s32_t)MAX_TICKS), 0);
 
 	k_spinlock_key_t key = k_spin_lock(&lock);
-	u32_t cyc, t = counter();
+	u32_t cyc, t2, t = counter();
 
 	/* Round up to next tick boundary */
-	cyc = ticks * CYC_PER_TICK + counter_sub(t, last_count);
+	cyc = ticks * CYC_PER_TICK + 1 + counter_sub(t, last_count);
```

Comment:
It's there for correctness.  If you pass in a ticks argument of zero, the API means "at the next tick" and not "immediately".   When ticks were longer than cycles, this didn't matter.  Now that we can trigger a synchronous interrupt, the distinction matters and this is here to avoid early timeouts.   Added a note in the commit message about it.